warp_drive/training/lightning_trainer.py

```diff
@@ -0,0 +1,996 @@
+# Copyright (c) 2021, salesforce.com, inc.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# For full license text, see the LICENSE file in the repo root
+# or https://opensource.org/licenses/BSD-3-Clause
+
+"""
+The PytorchLightning-based Trainer, PerfStats and Metrics classes
+"""
```

Comment:
Add a link to Pytorch Lightning and WarpDrive. Explain the context of this code.